test/LAMMPS_jll/.pkg/select_artifacts.jl

```diff
@@ -0,0 +1,27 @@
+# Note: This file is executed from `Pkg` in an isolated environment.
+#       We add the current package and stdlib to the load path so that
+#       `platform_augmentation.jl` can use the dependencies of the project,
+#       and `select_artifacts.jl` needs TOML, and Artifacts.
+# Warn: Since this is executed in a sandbox, the current preferences are not
+#       propagated and thus no decision can be made based on them.
+#       Use LazyArtifacts in this case.
+
+push!(Base.LOAD_PATH, "@stdlib")
+push!(Base.LOAD_PATH, dirname(@__DIR__))
+
+using TOML, Artifacts, Base.BinaryPlatforms
+include("./platform_augmentation.jl")
+artifacts_toml = joinpath(dirname(@__DIR__), "Artifacts.toml")
+
+# Get "target triplet" from ARGS, if given (defaulting to the host triplet otherwise)
+target_triplet = get(ARGS, 1, Base.BinaryPlatforms.host_triplet())
+
+# Augment this platform object with any special tags we require
+platform = augment_platform!(HostPlatform(parse(Platform, target_triplet)))
+
+# Select all downloadable artifacts that match that platform
+artifacts = select_downloadable_artifacts(artifacts_toml; platform, include_lazy = true)
```

Comment:
Because `Pkg` may execute this in a sandbox environment the include_lazy isn't that helpful, except with https://github.com/JuliaLang/Pkg.jl/pull/2901 (but maybe then we just don't need `select_artifacts.jl`)?

Reply:
What does `include_lazy` have to do with the current environment?  You're passing the `artifacts_toml` in directly, so this should be independent of whatever environment you have activated, right?

Reply:
This was about the fact that `augment_platform!` will not see current Preferences and thus won't do anything, but let's keep that discussion there.